src/main/java/com/google/api/gax/retrying/ScheduledRetryingExecutor.java

```diff
@@ -0,0 +1,101 @@
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.retrying;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The retry executor which uses {@link ScheduledExecutorService} to schedule an attempt tasks.
+ * Unless a direct executor service is used, this handler will schedule attempts for an execution in
+ * another thread.
+ *
+ * <p>
+ * This class is thread-safe.
+ *
+ * @param <ResponseT> response type
+ */
+public class ScheduledRetryingExecutor<ResponseT> implements RetryingExecutor<ResponseT> {
+
+  private final RetryAlgorithm retryAlgorithm;
+  private final ListeningScheduledExecutorService scheduler;
+
+  /**
+   * Creates a new scheduled retry executor, which will be using {@code scheduler} for actual
+   * attempts scheduling and {@code retryAlgorithm} for retrying strategy.
+   *
+   * @param retryAlgorithm retry algorithm to use
+   * @param scheduler scheduler
+   */
+  public ScheduledRetryingExecutor(
+      RetryAlgorithm retryAlgorithm, ScheduledExecutorService scheduler) {
+    this.retryAlgorithm = retryAlgorithm;
+    this.scheduler = MoreExecutors.listeningDecorator(scheduler);
+  }
+
+  /**
+   * Creates a {@link RetryingFuture}, which is a facade, returned to the client code to wait for
+   * any retriable operation to complete. The returned future is bounded to {@code this} executor
+   * instance.
+   *
+   * @param callable the actual callable, which should be executed in a retriable context
+   * @return retrying future facade
+   */
+  @Override
+  public RetryingFuture<ResponseT> createFuture(Callable<ResponseT> callable) {
+    return new RetryingFutureImpl<>(callable, retryAlgorithm, this);
+  }
+
+  /**
+   * Submits an attempt for execution in a different thread.
```

Comment:
I decided to remove it (I had it before) because, for example MoreExecutors has the "directExecutor()" version for a regular executor, but does not have one for ScheduledExecutorService. Probably because "scheduling" within same thread does not make much sense (it is rather "delaying" than "scheduling"). So I removed the notion of direct executor, because DirectScheduledExecutor is not a thing.